From 6ea8daccaf6746b1a4261a7c3390345223582eda Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Thu, 16 Jul 2020 17:46:06 +0800
Subject: [PATCH] Revert "drm: fix some no need warning"

This reverts commit 455796818d521a905c3fcb7621e0cc62a94bcd04.
and deal with this issue by following commit:
commit 1f525e2416b3 ("drm/rockchip: reset conn->state->best_encode")

Change-Id: I239296a0a618a3034c6502ca95050dd85da7b5fe
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/drm_atomic_helper.c | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/drm_atomic_helper.c b/drivers/gpu/drm/drm_atomic_helper.c
index 5adee7043f7a..fdad865d79f5 100644
--- a/drivers/gpu/drm/drm_atomic_helper.c
+++ b/drivers/gpu/drm/drm_atomic_helper.c
@@ -217,12 +217,8 @@ set_best_encoder(struct drm_atomic_state *state,
 		 * As an exception restoring duplicated atomic state
 		 * during resume is allowed, so don't warn when
 		 * best_encoder is equal to encoder we intend to set.
-		 *
-		 * As rockchip maybe appear one crtc connect two connecter,
-		 * the conn_state->best_encoder always isn't NULL, so here no
-		 * need warning.
 		 */
-		WARN_ON(!crtc && encoder && encoder != conn_state->best_encoder);
+		WARN_ON(!crtc && encoder != conn_state->best_encoder);
 		if (crtc) {
 			crtc_state = drm_atomic_get_new_crtc_state(state, crtc);
 
-- 
2.35.3

